dereference a binary type fails

The following program `sample.go` triggers a panic:

```go
package main

import (
    "fmt"
    "net"
    "strings"
)

type ipNetValue net.IPNet
    
func (ipnet *ipNetValue) Set(value string) error {
    _, n, err := net.ParseCIDR(strings.TrimSpace(value))
    if err != nil {
        return err
    }
    *ipnet = ipNetValue(*n)
    return nil
}

func main() {
    v := ipNetValue{}
    fmt.Println(v)
}
```

Expected result:
```console
$ go run ./sample.go
{<nil> <nil>}
```

Got:
```console
$ yaegi ./sample.go
scope.go:150: nil type
panic: nil type

goroutine 1 [running]:
log.Panic(0xc0001a52f8, 0x1, 0x1)
        /usr/lib/go/src/log/log.go:333 +0xac
github.com/containous/yaegi/interp.(*scope).add(0xc000144780, 0x0, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/scope.go:150 +0x1c2
github.com/containous/yaegi/interp.(*Interpreter).cfg.func2(0xc0001dd800)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1198 +0x43ae
github.com/containous/yaegi/interp.(*node).Walk(0xc0001dd800, 0xc0001a5d68, 0xc0001a5da0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:114 +0xaf
github.com/containous/yaegi/interp.(*node).Walk(0xc0001dd600, 0xc0001a5d68, 0xc0001a5da0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:111 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0001dd300, 0xc0001a5d68, 0xc0001a5da0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:111 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0001d9e00, 0xc0001a5d68, 0xc0001a5da0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:111 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0001d8f00, 0xc0001a5d68, 0xc0001a5da0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:111 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0001d8000, 0xc0001a5d68, 0xc0001a5da0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:111 +0x70
github.com/containous/yaegi/interp.(*Interpreter).cfg(0xc0001c6000, 0xc0001d8000, 0xc00014a0a0, 0x4, 0x0, 0x0, 0x4)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:39 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0001c6000, 0xc0001ce000, 0x133, 0x333, 0xc0001ce000, 0x133, 0x0, 0xc2f220)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:253 +0x135
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:77 +0x2f8

```

--- a/_test/ptr7.go
+++ b/_test/ptr7.go
@@ -1,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"strings"
+)
+
+type ipNetValue net.IPNet
+
+func (ipnet *ipNetValue) Set(value string) error {
+	_, n, err := net.ParseCIDR(strings.TrimSpace(value))
+	if err != nil {
+		return err
+	}
+	*ipnet = ipNetValue(*n)
+	return nil
+}
+
+func main() {
+	v := ipNetValue{}
+	fmt.Println(v)
+}
+
+// Output:
+// {<nil> <nil>}

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1202,7 +1202,11 @@ func (interp *Interpreter) cfg(root *node) ([]*node, error) {
 			default:
 				// dereference expression
 				wireChild(n)
-				n.typ = n.child[0].typ.val
+				if c0 := n.child[0]; c0.typ.cat == valueT {
+					n.typ = &itype{cat: valueT, rtype: c0.typ.rtype.Elem()}
+				} else {
+					n.typ = c0.typ.val
+				}
 				n.findex = sc.add(n.typ)
 			}
 

